lib/backend/watchersyncer/watchercache.go

```diff
@@ -201,11 +192,12 @@ func (wc *watcherCache) finishResync() {
 	if numOldResources > 0 {
 		wc.logger.WithField("Num", numOldResources).Debug("Sending resync deletes")
 		updates := make([]api.Update, 0, len(wc.oldResources))
-		for _, r := range wc.oldResources {
+		for p, _ := range wc.oldResources {
+			key := model.KeyFromDefaultPath(p)
```

Comment:
Worth asserting that key is not nil?  We should always be able to parse the keys we have here